server/ingest.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/verifier"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/go-chi/render"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func (a *applicationHandler) IngestEvent(w http.ResponseWriter, r *http.Request) {
+	// 1. Retrieve mask ID
+	maskID := r.URL.Query().Get("maskID")
+
+	// 2. Retrieve source using mask ID.
+	source, err := a.sourceRepo.FindSourceByMaskID(r.Context(), maskID)
```

Comment:
After retrieving the source, do we need to verify that the `source_type` is actually a `http` type to be able to proceed further?

Reply:
I think we do, because if it was changed verification will fail. On the flip side when updating sources the config should only be changed and not swapped out.